### ISSUE TITLE
Documenting Hakaru workflow

### ISSUE THREAD

**Issue by GenevaS** [hakaru-dev/hakaru#92]
We want to document (as tutorials) some of the problems that people want to solve and how they would do it using Hakaru.

**Comment by zaxtax** [comment]
That sounds like a good idea!

**Comment by GenevaS** [comment]
I have already created a page for the general workflow for Hakaru programming (7a849c332479e71bf6715d641ef3c7edebadeabf) and the burglary alarm example from the system description paper (6e257c8ee1b7332fd008ad8f2c72f69f27927553).

Which examples should I work on next?

**Comment by ccshan** [comment]
Thanks!

Regarding the burglary example, I would recommend (1) documenting the optional use of simplification (rather than sampling) after disintegration to figure an exact (rather than approximate) answer to the problem; (2) add each transformation in the documented workflow as tests.

Regarding which example next, I would recommend the second of the two examples of the system description paper, which we call "easier roadmap".

**Comment by GenevaS** [comment]
I've added simplification to the discrete model workflow (fc2af243d90015022ec2313e45bb869456786bb5) and I will start work on the continuous model.

**Comment by GenevaS** [comment]
As of 45456a14a78f87221a376ed043feec04d8dfcccc, the continuous model/easy road workflow has been documented and added to the test suite.

**Comment by GenevaS** [comment]
I have been reading the [paper](https://arxiv.org/pdf/1603.01882.pdf)  that was suggested to me for documenting GMM Gibbs, and I have a few questions:

1. How did you want this example to look in the documentation? It is not like the burglary or thermometer examples (which were published like tutorials). This one looks more like a "Sure, you can do this in Hakaru". My only thought right now is that perhaps this could be added to the "Examples" page with a description of the program and a reference to the paper. If you would prefer a page on its own, what should be included? Regarding testing, this pipeline has already been added to ``haskell/Tests/RoundTrip.hs``.

2. The first example in the paper is a series of transformations on a Kalman filter. Isn't this the same as the thermometer example (hakaru/tests/RoundTrip/thermometer_workflow.hk)? If it isn't, I want to add it to ``haskell/Tests/RoundTrip.hs`` as a pipeline.

**Comment by ccshan** [comment]
Is this done?

**Comment by JacquesCarette** [comment]
I am not actually 100% sure that this is done -- Geneva was waiting for Yuriy to finish a couple of things to close this off, and he didn't finish them until some time in September.

**Comment by GenevaS** [comment]
No, this issue has not yet been resolved. I am reopening it, but I will not be able to work on it for a few weeks yet.
